### PR TITLE
audit: Increase speed when using socket_events

### PR DESCRIPTION
**Pull request by theopolis**
There are several things here that attempt to make process and socket auditing faster for Linux.

New 'hidden' flags for testing audit:
- `audit_debug` turning this on will print the audit messages to `stdout`.
- `rocksdb_write_buffer` controls the max write buffer
- `rocksdb_merge_number` controls the min write buffer to merge
- `rocksdb_background_flushes` controls the number of background flush threads
- `rocksdb_buffer_blocks` controls the number of 4kB write blocks

These flags allow us to tune RocksDB in performance tests. Assume the defaults are the tested-best. 

The socket auditing in Linux is limited with this PR. The UNIX sockets are removed by default. If you would like to include UNIX socket auditing use `--audit_allow_unix`.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 3d10458cd9da8817ce3002960ecccb57631cf4f0 (Job results: [1954](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1954/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 3d10458cd9da8817ce3002960ecccb57631cf4f0 (Job results: [5200](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5200/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 3d10458cd9da8817ce3002960ecccb57631cf4f0 (Job results: [5201](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5201/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 3d10458cd9da8817ce3002960ecccb57631cf4f0 (Job results: [2098](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2098/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 3d10458cd9da8817ce3002960ecccb57631cf4f0 (Job results: [2099](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2099/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit c98a94bd12e93725d3dc8faef86da81837034cca (Job results: [2100](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2100/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit c98a94bd12e93725d3dc8faef86da81837034cca (Job results: [2101](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2101/)) failed one or more tests (**Windows**).

File: osquery/events/linux/audit.cpp [hunk 1]
```diff
@@ -355,6 +357,14 @@ bool handleAuditReply(const struct audit_reply& reply,
     ec->fields.emplace(std::make_pair(std::move(key), std::move(value)));
   }
 
+  if (FLAGS_audit_debug) {
+    printf("%zu: (%d) ", ec->audit_id, ec->type);
```
Comment (theopolis) [inline osquery/events/linux/audit.cpp]: To help with `printf` scanning, lets move these to `fprintf`.

File: osquery/tables/events/linux/socket_events.cpp [hunk 2]
```diff
@@ -28,6 +28,11 @@ FLAG(bool,
      false,
      "Allow the audit publisher to install socket-related rules");
 
+HIDDEN_FLAG(bool,
+            audit_allow_unix,
```
Comment (theopolis) [inline osquery/tables/events/linux/socket_events.cpp]: This needs documentation too.

**Pull request merged**